### PR TITLE
chore: migrate to husky@8

### PR DESCRIPTION
**Pull request by 0618**
#### Description of changes

After https://github.com/aws-amplify/amplify-cli/pull/12869, it has some issues:
1. error message in exist local folder
> Can't find Husky, skipping post-commit hook
> You can reinstall it using 'npm install husky --save-dev' or delete this hook

2. if clone a new repo, the hook would not work

**Reason**: Husky doesn't use config in package.json after version 7 ([ref](https://stackoverflow.com/a/68761163/12610324))
**Fix**: This PR is to migrate the setup to make husky@8 works. [Migration Guide](https://typicode.github.io/husky/getting-started.html)

#### Description of how you validated changes

image

image

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -81,13 +82,6 @@
     "type": "git",
     "url": "https://github.com/aws-amplify/amplify-cli.git"
   },
-  "husky": {
-    "hooks": {
-      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS",
```
Comment (0618) [inline package.json:86]: Husky no longer uses `HUSKY_GIT_PARAMS` since v5 [ref](https://stackoverflow.com/a/57864128/12610324)

File: package.json [hunk 2]
```diff
@@ -81,13 +82,6 @@
...
-      "pre-push": "yarn build-tests-changed && yarn split-e2e-tests",
-      "pre-commit": "yarn prettier-changes && yarn lint-fix && ./scripts/package-json-check.sh && yarn verify-commit"
```
Comment (0618) [inline package.json:88]: We don't use `prettier-changes` because:
1. we need to change the staged files, not all changed files
2. `prettier-changes` doesn't stage after changing it

File: package.json [hunk 3]
```diff
@@ -69,7 +69,8 @@
     "verify-api-extract": "yarn extract-api && ./scripts/verify-extract-api.sh",
     "verify-commit": "yarn ts-node ./scripts/verify-commit.ts",
     "verify-yarn-lock": "./scripts/verify-yarn-lock.sh",
-    "workflow-results": "yarn ts-node ./scripts/cci-workflow-results.ts"
+    "workflow-results": "yarn ts-node ./scripts/cci-workflow-results.ts",
+    "postinstall": "husky install"
```
Comment (jhockett) [inline package.json:73]: nit: sorting

File: scripts/package-json-check.sh [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
 if [ ! -z "git diff --diff-filter=MRA --name-only | grep package.json" ]; then
   yarn lint-fix-package-json
-  echo "Changes made to package.json files. Please stage these changes before pushing."
+  git diff --diff-filter=MRA --name-only | grep package.json | xargs git add
```
Comment (jhockett) [inline scripts/package-json-check.sh:4]: Is this related to the husky fix?

**Review by jhockett — COMMENTED** [review]
nit: please merge this as a `chore` conventional commit instead of `fix` because it doesn't impact consumers of Amplify CLI

File: scripts/package-json-check.sh (hunk 4 above)
Comment (0618) [reply scripts/package-json-check.sh:4]: Yes, previously this file only lint, but don't stage the change, which means, you have to edit the commit or make another commit for the change just made by lint

File: package.json (hunk 3 above)
Comment (0618) [reply package.json]: done 3524fb7d75a7f80a4a16d99a939dd66bc6d37088

**Comment by 0618** [comment]
> nit: please merge this as a `chore` conventional commit instead of `fix` because it doesn't impact consumers of Amplify CLI

done

**Review by jhockett — APPROVED** [review]

**Review by sobolk — APPROVED** [review]
